app/Models/Semester.php

```diff
@@ -258,29 +258,23 @@ public function questionsNotAnsweredBy(User $user)
     }
 
     /**
-     * Returns the current semester from cache.
      * There is always a "current" semester. If there is not in the database, this function creates it.
      * In case the current time is in between two semesters, it is still undefined as we follow the months and not the getEndDate/getStartDate.
      */
     public static function current(): Semester
     {
         $today = Carbon::today()->format('Ymd');
-        if (!Cache::get('semester.current.' . $today)) {
-            $now = Carbon::now();
-            if ($now->month >= self::START_OF_SPRING_SEMESTER && $now->month <= self::END_OF_SPRING_SEMESTER) {
-                $part = 2;
-                $year = $now->year - 1;
-            } else {
-                $part = 1;
-                // This assumes that the semester ends in the new year.
-                $year = $now->month <= self::END_OF_AUTUMN_SEMESTER ? $now->year - 1 : $now->year;
-            }
-            $current = Semester::getOrCreate($year, $part);
-
-            Cache::put('semester.current.' . $today, $current, Carbon::tomorrow());
+        $now = Carbon::now();
+        if ($now->month >= self::START_OF_SPRING_SEMESTER && $now->month <= self::END_OF_SPRING_SEMESTER) {
+            $part = 2;
+            $year = $now->year - 1;
+        } else {
+            $part = 1;
+            // This assumes that the semester ends in the new year.
+            $year = $now->month <= self::END_OF_AUTUMN_SEMESTER ? $now->year - 1 : $now->year;
         }
-
-        return Cache::get('semester.current.' . $today);
+        $current = Semester::getOrCreate($year, $part);
+        return $current;
```

Comment:
why are you deleting caching?

Reply:
I forgot to remove it before pushing. It was a pain to debug my changes with rapid fresh generations of the database and having cache (the cache remained intact while I executed `php artisan migrate:fresh --seed`).

I was wondering in the last couple of days how much real world benefits does caching have in similar instances, but sometimes it makes debugging harder (cache integrity has bit me just a couple of days ago 😅 I just started disabling caching temporarily when I suspect that it might interfere with my debugging and re-enable afterwards). But I did not want to bring it to this PR.